usersession/userd/privileged_desktop_launcher_internal_test.go

```diff
@@ -0,0 +1,486 @@
+		{"foo-bar-baz.desktop", "/var/lib/snapd/desktop/applications/foo-bar/baz.desktop"},
+		{"baz-foo-bar.desktop", "/var/lib/snapd/desktop/applications/baz/foo-bar.desktop"},
+	}
+
+	for _, test := range desktopIdTests {
+		actual, err := userd.DesktopFileIDToFilename(test.id)
+		c.Assert(err, IsNil)
+		c.Assert(actual, Equals, test.expect)
+	}
+}
+
+func (s *privilegedDesktopLauncherInternalSuite) TestDesktopFileIDToFilenameFailsWithInvalidId(c *C) {
+	restore := userd.MockRegularFileExists(existsOnMockFileSystem)
+	defer restore()
+
+	var desktopIdTests = []string{
+		"mir-kiosk-scummvm-mir-kiosk-scummvm.desktop",
+		"bar-foo-baz.desktop",
+		"bar-baz-foo.desktop",
+		"foo-bar_foo-bar.desktop",
+	}
+
+	for _, id := range desktopIdTests {
+		_, err := userd.DesktopFileIDToFilename(id)
+		c.Assert(err, NotNil)
+	}
+}
+
+func (s *privilegedDesktopLauncherInternalSuite) TestParseExecCommandSucceedsWithValidEntry(c *C) {
+	var exec_command = []struct {
+		exec_command string
+		expect       []string
+	}{
+		// valid with no exec variables
+		{"env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/mir-kiosk-scummvm_mir-kiosk-scummvm.desktop /snap/bin/mir-kiosk-scummvm %U",
+			[]string{"env", "BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/mir-kiosk-scummvm_mir-kiosk-scummvm.desktop", "/snap/bin/mir-kiosk-scummvm"}},
+		// valid with literal '%' and no exec variables
+		{"/snap/bin/foo -f %%bar", []string{"/snap/bin/foo", "-f", "%bar"}},
+		{"/snap/bin/foo -f %%bar %%baz", []string{"/snap/bin/foo", "-f", "%bar", "%baz"}},
+		// valid where quoted strings are passed through
+		{"/snap/bin/foo '-f %U'", []string{"/snap/bin/foo", "-f %U"}},
+		{"/snap/bin/foo '-f %%bar'", []string{"/snap/bin/foo", "-f %%bar"}},
+		{"/snap/bin/foo '-f %U %%bar'", []string{"/snap/bin/foo", "-f %U %%bar"}},
+		{"/snap/bin/foo \"'-f bar'\"", []string{"/snap/bin/foo", "'-f bar'"}},
+		{"/snap/bin/foo '\"-f bar\"'", []string{"/snap/bin/foo", "\"-f bar\""}},
+		// valid with exec variables stripped out
+		{"/snap/bin/foo -f %U", []string{"/snap/bin/foo", "-f"}},
+		{"/snap/bin/foo -f %U %i", []string{"/snap/bin/foo", "-f", "--icon", "/snap/chromium/1193/chromium.png"}},
+		{"/snap/bin/foo -f %U bar", []string{"/snap/bin/foo", "-f", "bar"}},
+		{"/snap/bin/foo -f %U bar %i", []string{"/snap/bin/foo", "-f", "bar", "--icon", "/snap/chromium/1193/chromium.png"}},
+		// valid with mixture of literal '%' and exec variables
+		{"/snap/bin/foo -f %U %%bar", []string{"/snap/bin/foo", "-f", "%bar"}},
+		{"/snap/bin/foo -f %U %i %%bar", []string{"/snap/bin/foo", "-f", "--icon", "/snap/chromium/1193/chromium.png", "%bar"}},
+		{"/snap/bin/foo -f %U %%bar %i", []string{"/snap/bin/foo", "-f", "%bar", "--icon", "/snap/chromium/1193/chromium.png"}},
+		{"/snap/bin/foo -f %%bar %U %i", []string{"/snap/bin/foo", "-f", "%bar", "--icon", "/snap/chromium/1193/chromium.png"}},
```

Comment:
could you also add some more nonsensical fuzzing cases like 
- [ ] "/snap/bin/foo %i %i %i %i %i %i"
- [ ] "/snap/bin/%i foo bar"
- [ ] "%i /snap/bin/foo bar %"
- [ ] "%i/snap/bin/foobar %i"

Reply:
this is okay to do as a followup